### ISSUE TITLE
Editing Node Label Causes Rancher Panic 

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#27181]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug only in master-head, not in v2.4-head

**Steps to reproduce (least amount of steps as possible):**
- run Rancher:master-head b85ea695f, single install
- add a custom cluster with node labels (under `show advanced option` in the Node Options section)
- the cluster is up and active, nodes get the labels 
- edit a node in Rancher UI to change the value of an existing label

**Result:**
- Rancher server crashes with the following error in logs

`E0521 00:00:02.963794       6 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)`

```
E0521 00:00:02.963794       6 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 2418 [running]:
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x3676600, 0x6bad200)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x3676600, 0x6bad200)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/rancher/rancher/pkg/controllers/user/nodesyncer.(*nodesSyncer).syncTaints(0xc004576690, 0xc000bceea0, 0x16, 0xc00a51c000, 0xc000077c00, 0x7f5be096b6d0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/pkg/controllers/user/nodesyncer/nodetaints.go:32 +0x123
github.com/rancher/rancher/vendor/github.com/rancher/types/apis/management.cattle.io/v3.(*nodeController).AddHandler.func1(0xc000bceea0, 0x16, 0x3dcd180, 0xc00a51c000, 0xc009de7ad0, 0x3, 0x3, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/types/apis/management.cattle.io/v3/zz_generated_node_controller.go:152 +0x63
github.com/rancher/rancher/vendor/github.com/rancher/norman/controller.(*genericController).AddHandler.func1(0xc000bceea0, 0x16, 0x464aac0, 0xc00a51c000, 0x464aac0, 0xc00a51c000, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:57 +0x1e1
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.SharedControllerHandlerFunc.OnChange(0xc007f58810, 0xc000bceea0, 0x16, 0x464aac0, 0xc00a51c000, 0x464aac0, 0xc00a51c000, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/sharedcontroller.go:29 +0x4e
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*sharedHandler).OnChange(0xc000f22de0, 0xc000bceea0, 0x16, 0x464aac0, 0xc00a51c000, 0x42e801, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/sharedhandler.go:65 +0x150
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*controller).syncHandler(0xc00092bce0, 0xc000bceea0, 0x16, 0xc002be15c0, 0xc002f3ad30)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/controller.go:211 +0xd1
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*controller).processSingleItem(0xc00092bce0, 0x33ba480, 0xc000112110, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/controller.go:193 +0xf0
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*controller).processNextWorkItem(0xc00092bce0, 0x40c486)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/controller.go:170 +0x51
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*controller).runWorker(0xc00092bce0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/controller.go:159 +0x2b
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc002ad5030)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5e
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc002ad5030, 0x460ebc0, 0xc002a53c50, 0xc002bc8201, 0xc000c0aea0)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xa3
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc002ad5030, 0x3b9aca00, 0x0, 0x1, 0xc000c0aea0)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0xe2
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc002ad5030, 0x3b9aca00, 0xc000c0aea0)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*controller).run
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/controller.go:130 +0x35e
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x168 pc=0x23c9983]
```

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): master-head b85ea695f
- Installation option (single install/HA): single install 

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory): ec2 instance t2.medium 
- Kubernetes version (use `kubectl version`):

```
 1.17.5
```

- Docker version (use `docker version`):

```
Client:
 Version:           18.06.3-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        d7080c1
 Built:             Wed Feb 20 02:28:10 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.3-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       d7080c1
  Built:            Wed Feb 20 02:26:34 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Comment by sowmyav27** [comment]
Hit this issue on latest master-head commit id: `9f6da9ed1` while adding annotations on the node. The rancher server keeps restarting.

**Comment by sowmyav27** [comment]
**on master-head commit id: `fd3e37bec`**

- bring up a cluster (AWS cluster 1 master/control plane and 3 worker nodes)
- Add/delete labels/annotations on a worker node
- No panic is seen in the rancher server logs

**Comment by sowmyav27** [comment]
**Reopening the issue as we see the same issue on master-head - commit id: `3d7c3b3c1`**

```
E0804 22:11:59.095151       7 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 6854 [running]:
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x3b176e0, 0x74cd350)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x3b176e0, 0x74cd350)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/rancher/rancher/pkg/controllers/managementuser/nodesyncer.(*nodesSyncer).syncTaints(0xc008cacee0, 0xc002d5cfa0, 0x16, 0xc006e0fb00, 0xc0031cea80, 0x7fda5bac66d0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/pkg/controllers/managementuser/nodesyncer/nodetaints.go:34 +0x123
github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3.(*nodeController).AddHandler.func1(0xc002d5cfa0, 0x16, 0x4335940, 0xc006e0fb00, 0xc002bbdad0, 0x3, 0x3, 0x0)
	/go/src/github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3/zz_generated_node_controller.go:150 +0x63
github.com/rancher/rancher/vendor/github.com/rancher/norman/controller.(*genericController).AddHandler.func1(0xc002d5cfa0, 0x16, 0x4c8a060, 0xc006e0fb00, 0x4c8a060, 0xc006e0fb00, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:57 +0x1e1
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.SharedControllerHandlerFunc.OnChange(0xc00351bec0, 0xc002d5cfa0, 0x16, 0x4c8a060, 0xc006e0fb00, 0x4c8a060, 0xc006e0fb00, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/sharedcontroller.go:29 +0x4e
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*sharedHandler).OnChange(0xc0011512c0, 0xc002d5cfa0, 0x16, 0x4c8a060, 0xc006e0fb00, 0x42e901, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/sharedhandler.go:65 +0x150
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*controller).syncHandler(0xc0009680b0, 0xc002d5cfa0, 0x16, 0xc005461b30, 0xc002bbdd30)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/controller.go:211 +0xd1
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*controller).processSingleItem(0xc0009680b0, 0x3815c20, 0xc00b565100, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/controller.go:193 +0xf0
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*controller).processNextWorkItem(0xc0009680b0, 0x40c501)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/controller.go:170 +0x51
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*controller).runWorker(0xc0009680b0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/controller.go:159 +0x2b
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0029f6ed0)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5e
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0029f6ed0, 0x4c4aae0, 0xc003434570, 0xc0029e4f01, 0xc0022b7740)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xa3
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0029f6ed0, 0x3b9aca00, 0x0, 0x1, 0xc0022b7740)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0xe2
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc0029f6ed0, 0x3b9aca00, 0xc0022b7740)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*controller).run
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/controller.go:130 +0x35e
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x168 pc=0x2728c83]

goroutine 6854 [running]:
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x105
panic(0x3b176e0, 0x74cd350)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/rancher/rancher/pkg/controllers/managementuser/nodesyncer.(*nodesSyncer).syncTaints(0xc008cacee0, 0xc002d5cfa0, 0x16, 0xc006e0fb00, 0xc0031cea80, 0x7fda5bac66d0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/pkg/controllers/managementuser/nodesyncer/nodetaints.go:34 +0x123
github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3.(*nodeController).AddHandler.func1(0xc002d5cfa0, 0x16, 0x4335940, 0xc006e0fb00, 0xc002bbdad0, 0x3, 0x3, 0x0)
	/go/src/github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3/zz_generated_node_controller.go:150 +0x63
github.com/rancher/rancher/vendor/github.com/rancher/norman/controller.(*genericController).AddHandler.func1(0xc002d5cfa0, 0x16, 0x4c8a060, 0xc006e0fb00, 0x4c8a060, 0xc006e0fb00, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:57 +0x1e1
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.SharedControllerHandlerFunc.OnChange(0xc00351bec0, 0xc002d5cfa0, 0x16, 0x4c8a060, 0xc006e0fb00, 0x4c8a060, 0xc006e0fb00, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/sharedcontroller.go:29 +0x4e
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*sharedHandler).OnChange(0xc0011512c0, 0xc002d5cfa0, 0x16, 0x4c8a060, 0xc006e0fb00, 0x42e901, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/sharedhandler.go:65 +0x150
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*controller).syncHandler(0xc0009680b0, 0xc002d5cfa0, 0x16, 0xc005461b30, 0xc002bbdd30)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/controller.go:211 +0xd1
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*controller).processSingleItem(0xc0009680b0, 0x3815c20, 0xc00b565100, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/controller.go:193 +0xf0
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*controller).processNextWorkItem(0xc0009680b0, 0x40c501)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/controller.go:170 +0x51
github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*controller).runWorker(0xc0009680b0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/controller.go:159 +0x2b
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0029f6ed0)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5e
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0029f6ed0, 0x4c4aae0, 0xc003434570, 0xc0029e4f01, 0xc0022b7740)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xa3
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0029f6ed0, 0x3b9aca00, 0x0, 0x1, 0xc0022b7740)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0xe2
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc0029f6ed0, 0x3b9aca00, 0xc0022b7740)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller.(*controller).run
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/lasso/pkg/controller/controller.go:130 +0x35e
```

**Comment by dramich** [comment]
PR: https://github.com/rancher/rancher/pull/28591

**Comment by jiaqiluo** [comment]
The origin bug is fixed on `master-4149ea9e955b6989dee3bc696125e4e714807b67-head` 

Steps are the same as the issue provides. 

Now we can edit the labels on nodes and NO panic in Rancher's logs.

A new issue is found and tracked here: https://github.com/rancher/rancher/issues/28617
